samples/dnn/face_detect.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+
+import cv2 as cv
+import numpy as np
+
+def str2bool(v):
+    if v.lower() in ['on', 'yes', 'true', 'y', 't']:
+        return True
+    elif v.lower() in ['off', 'no', 'false', 'n', 'f']:
+        return False
+    else:
+        raise NotImplementedError
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--input', '-i', type=str, help='Path to the input image.')
+parser.add_argument('--model', '-m', type=str, default='yunet.onnx', help='Path to the model. Download the model at https://github.com/ShiqiYu/libfacedetection.train/tree/master/tasks/task1/onnx.')
+parser.add_argument('--score_threshold', type=float, default=0.9, help='Filtering out faces of score < score_threshold.')
+parser.add_argument('--nms_threshold', type=float, default=0.3, help='Suppress bounding boxes of iou >= nms_threshold.')
+parser.add_argument('--top_k', type=int, default=5000, help='Keep top_k bounding boxes before NMS.')
+parser.add_argument('--save', '-s', type=str, default=False, help='Set true to save results. This flag is invalid when using camera.')
+parser.add_argument('--vis', '-v', type=str2bool, default=True, help='Set true to open a window for result visualization. This flag is invalid when using camera.')
+args = parser.parse_args()
+
+def visualize(input, faces, thickness=2):
+    output = input.copy()
+    if faces[1] is not None:
+        for idx, face in enumerate(faces[1]):
+            print('Face {}, top-left coordinates: ({:.0f}, {:.0f}), box width: {:.0f}, box height {:.0f}, score: {:.2f}'.format(idx, face[0], face[1], face[2], face[3], face[-1]))
+
+            coords = face[:-1].astype(np.int32)
+            cv.rectangle(output, (coords[0], coords[1]), (coords[0]+coords[2], coords[1]+coords[3]), (0, 255, 0), 2)
+            cv.circle(output, (coords[4], coords[5]), 2, (255, 0, 0), 2)
+            cv.circle(output, (coords[6], coords[7]), 2, (0, 0, 255), 2)
+            cv.circle(output, (coords[8], coords[9]), 2, (0, 255, 0), 2)
+            cv.circle(output, (coords[10], coords[11]), 2, (255, 0, 255), 2)
+            cv.circle(output, (coords[12], coords[13]), 2, (0, 255, 255), 2)
+    return output
+
+if __name__ == '__main__':
+
+    # Instantiate FaceDetectorYN
+    detector = cv.FaceDetectorYN.create(
+        args.model,
+        "",
+        (320, 320),
```

Comment:
> (320, 320)

How is this aligned with .setInputSize()?

Reply:
This algorithm requires to generate priors/anchors for different input sizes. My intention is to allow users to set the input size for the network when creating instances in case the input size is fixed, such as taking video stream as input. However, users may also want to detect on images of different sizes with one instance, so `.setInputSize()` is added to do so.